operator/src/main/java/org/keycloak/operator/controllers/KeycloakIngress.java

```diff
@@ -114,6 +115,22 @@ protected void deleteExistingIngress() {
         client.network().v1().ingresses().inNamespace(getNamespace()).delete(existingIngress);
     }
 
+    private boolean isExistingIngressFromSameOwnerReference() {
+
+        if ((Objects.nonNull(this.keycloak) && Objects.nonNull(this.keycloak.getMetadata().getUid())) &&
+            (Objects.nonNull(this.existingIngress) && Objects.nonNull(this.existingIngress.getMetadata().getOwnerReferences()))) {
```

Comment:
I only added this ```If``` because the error below happens in this other test:

`
org.opentest4j.AssertionFailedError: 
Expected :true
Actual   :false
<Click to see difference>
        ...
	at org.keycloak.operator.testsuite.unit.IngressLogicTest.testIngressDisabledExisting(IngressLogicTest.java:99)
        ...
`

The `deleteExistingIngress()` method must run for that the `IngressLogicTest.testIngressDisabledExisting()` can perform successfully.

When running the `IngressLogicTest.testIngressDisabledExisting()`, it is created a Mocked Ingress with its `Metadata` as `null`. And yet still the `deleteExistingIngress()` method must be run.

So I'm gonna keep that `If` to have all tests passing OK.

WDYT, @vmuzikar ?

Reply:
I only added this ```If``` because the error below happens in this other test:

`
org.opentest4j.AssertionFailedError: 
Expected :true
Actual   :false
<Click to see difference>
        ...
	at org.keycloak.operator.testsuite.unit.IngressLogicTest.testIngressDisabledExisting(IngressLogicTest.java:99)
        ...
`

The `deleteExistingIngress()` method must run for that the `IngressLogicTest.testIngressDisabledExisting()` can perform successfully.

When running the `IngressLogicTest.testIngressDisabledExisting()`, it is created a Mocked Ingress with its `Metadata` as `null`. And yet still the `deleteExistingIngress()` method must be run.

So I'm gonna keep that `If` to have all tests passing OK.

WDYT, @vmuzikar ?

Reply:
I would fix the `testIngressDisabledExisting` to add (empty) metadata. In real-life scenarios, we can expect the metadata for existing Ingress are there. Otherwise, something's wrong and the reconciliation should fail.